python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -559,25 +602,107 @@ cdef class Column:
 
+    @staticmethod
+    cdef Column _from_gpumemoryview(
+        gpumemoryview data,
+        tuple shape,
+        tuple strides,
+        DataType dtype,
+    ):
+        """
+        Construct a Column from a gpumemoryview and array metadata.
+
+        Note: This method is called with ``_validate_array_metadata``
+        in the relevant methods.
+        """
+        if len(shape) == 1:
+            size = shape[0]
+            return Column(dtype, size, data, None, 0, 0, [])
+
+        else:
+            num_rows, num_cols = shape
+
+            offsets_col = sequence(
+                num_rows + 1,
+                Scalar.from_py(0, DataType(type_id.INT32)),
+                Scalar.from_py(num_cols, DataType(type_id.INT32)),
+            )
+
+            flat_size = num_rows * num_cols
+
+            data_col = Column(
+                data_type=dtype,
+                size=flat_size,
+                data=data,
+                mask=None,
+                null_count=0,
+                offset=0,
+                children=[],
+            )
+
+            return Column(
+                data_type=DataType(type_id.LIST),
+                size=num_rows,
+                data=None,
+                mask=None,
+                null_count=0,
+                offset=0,
+                children=[offsets_col, data_col],
+            )
+
     @classmethod
     def from_array_interface(cls, obj):
         """
         Create a Column from an object implementing the NumPy Array Interface.
 
+        If the object provides a raw memory pointer via the "data" field,
+        we use that pointer directly and avoid copying. Otherwise, a ValueError
+        is raised.
+
         Parameters
         ----------
         obj : object
-            Must implement the `__array_interface__` protocol.
+            Must implement the ``__array_interface__`` protocol.
+
+        Returns
+        -------
+        Column
+            A Column containing the data from the array interface.
 
         Raises
         ------
+        TypeError
+            If the object does not implement ``__array_interface__``.
+        ValueError
+            If the array is not 1D or 2D, or is not C-contiguous.
+            If the number of rows exceeds size_type limit.
+            If the 'data' field is invalid.
         NotImplementedError
-            This method is not yet implemented.
+            If the object has a mask.
         """
-        raise NotImplementedError(
-            "Converting to a pylibcudf Column is not yet implemented."
+        try:
+            iface = obj.__array_interface__
+        except AttributeError:
+            raise TypeError("Object does not implement __array_interface__")
+
+        typestr, data, shape, strides = _parse_array_interface(iface)
+        dtype = _datatype_from_dtype_desc(typestr[1:])
+        itemsize = size_of(dtype)
+        _validate_array_metadata(typestr, itemsize, shape, data, strides)
```

Comment:
validation and parsing could be put together so that we get back:

```
cdef uintptr_t data_ptr
nbytes, dtype, shape, strides, data_ptr = _parse_array_interface(iface)
dbuf = DeviceBuffer.to_device(...)
return Column._from_gpumemoryview(gpumemoryview(dbug), shape, strides, dtype)
```